taxi/views.py

```diff
@@ -31,11 +39,26 @@ def index(request):
 
 
 class ManufacturerListView(LoginRequiredMixin, generic.ListView):
-    model = Manufacturer
-    context_object_name = "manufacturer_list"
-    template_name = "taxi/manufacturer_list.html"
     paginate_by = 5
 
+    def get_queryset(self):
+        name = self.request.GET.get("name")
+        queryset = Manufacturer.objects.all()
+
+        if name:
+            return queryset.filter(name__icontains=name)
+        return queryset
```

Comment:
It would be more logical to use "name__startswith=name" lookup here
If I get this homework rejected, I will change it for myself too :)
Just telling